### ISSUE TITLE
Destroyed pulse still called once (I think?) if replaced via AJAX

### ISSUE THREAD

**Issue by parryc** [jsoverson/jquery.pulse.js#3]
The basic setup I have is that depending on parameters you set, an AJAX call will generate new output (completely replacing old output), which starts to pulse accordingly. From a user's perspective everything works fine, elements load, start pulsing, stop with user input, etc.

However, if you look at the console, everytime there's a call it's filled with `TypeError: 'undefined' is not an object (evaluating 'el.data("pulse").stop')`.  I've thrown in a beforeSend function with the AJAX that ostensibly kills everything, but it doesn't seem to make the error go away.  

Any thoughts?

**Comment by geilt** [comment]
More than likely the issue is since your destroying the object, you also destroy what it is bound to, since it looks like you are setting your jQuery selectors to variables, the memory reference to that item is destroyed, then not exactly replaced, but you are calling stop on it so it is trying to do something to it.

Would love to see an example as to what exactly you are doing. 

Regards,

**Comment by parryc** [comment]
All of the work is going on here.  I've removed a lot of the extraneous code. 

``` javascript
//Generate new marked output on click.  Replace old output
$('#send').click(function(){
    $.ajax({url: 'mark.php',
        type: 'post',
        beforeSend: destroyPulses,
        dataType: 'json',
        data: {...},
        success: function(response){
            var usedInputs = getUsedInputs(),
                times = usedInputs.length,
                count = 0;
            $('#output').html(response);
            startPulse(times, usedInputs, count);
    });
});

//Start a cycle of pulses, so that not every color is blinking in unison
function startPulse(times,usedInputs,count){
    setTimeout(function () {
          var num = usedInputs[count],
            color = getColor(num);
            getMarkedByNumber(num).pulse({ color : color, opacity: 0},{pulses: -1, duration: 1000, interval: (usedInputs.length*1000)});
           count++;
           if (count < times)
            startPulse(times,usedInputs,count);
        }, 1000);
}

//Detroy all pulses, regardless of color or input
function destroyPulses(){
    return $('.marked').pulse('destroy');
}
```

**Comment by jsoverson** [comment]
Closed via PR #4
